### PR TITLE
[6.5] UI Port role tests

### PR DESCRIPTION
**Pull request by ldjebran**
closes https://github.com/SatelliteQE/robottelo/issues/6395

dependencies:
 -  https://github.com/SatelliteQE/airgun/pull/290
 -  https://github.com/SatelliteQE/airgun/pull/292
```
(robottelo) dlezz@dlezz:~/projects/robottelo-fork$ pytest -v tests/foreman/ui_airgun/test_role.py
============================================ test session starts =============================================
platform linux -- Python 3.6.6, pytest-4.0.2, py-1.5.4, pluggy-0.8.0 -- /home/dlezz/.pyenv/versions/robottelo/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - ON - scope: ffff - storage: redis
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.26.0, timeout-1.3.3, services-1.3.1, mock-1.10.0, forked-0.2, cov-2.6.1
collecting ... 2019-02-19 15:03:04 - conftest - DEBUG - BZ deselect is disabled in settings

collected 11 items                                                                                           

tests/foreman/ui_airgun/test_role.py::test_positive_end_to_end PASSED                                  [  9%]
tests/foreman/ui_airgun/test_role.py::test_positive_assign_cloned_role PASSED                          [ 18%]
tests/foreman/ui_airgun/test_role.py::test_positive_delete_cloned_builtin PASSED                       [ 27%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_filter_without_override PASSED              [ 36%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_non_overridable_filter PASSED               [ 45%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_overridable_filter PASSED                   [ 54%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_with_21_filters PASSED                      [ 63%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_with_sc_parameter_permission PASSED         [ 72%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_with_smart_variable_permission PASSED       [ 81%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_filter_admin_user_with_locs PASSED          [ 90%]
tests/foreman/ui_airgun/test_role.py::test_positive_create_filter_admin_user_with_orgs PASSED          [100%]

================================== 11 passed, 6 warnings in 1292.57 seconds ==================================

```

### CODE REVIEW

File: tests/foreman/ui_airgun/test_role.py [hunk 1]
```diff
@@ -33,6 +36,132 @@ def module_loc():
...
+
+    :CaseImportance: Critical
+    """
+    role_name = random.choice(ROLES)
+    cloned_role_name = gen_string('alpha')
+    user_name = gen_string('alpha')
+    user_password = gen_string('alpha')
+    with session:
+        session.role.clone(role_name, {'name': cloned_role_name})
+        assert session.role.search(cloned_role_name)[0]['Name'] == cloned_role_name
```
Comment (abalakh) [inline tests/foreman/ui_airgun/test_role.py]: We can make this scenario better - not just ensure some role with proper name was created (such scenario isn't that different from `create` test), but read values in original role, then clone, read cloned values, ensure they're the same. Such flow will verify cloning actually working.

Comment (ldjebran) [reply tests/foreman/ui_airgun/test_role.py]: The clone work only for filters (create a new role with cloning the filters), new values should be entered for the clone.

Comment (abalakh) [reply tests/foreman/ui_airgun/test_role.py]: Well, you got my point - if only filters are cloned, then check filters instead. Idea is to ensure cloning works, not just some new empty entity was created.

Comment (ldjebran) [reply tests/foreman/ui_airgun/test_role.py]: @abalakh This is a bug coverage test (that has a concrete scenario), there is other test where we verify that the cloned role has that permissions.

**Pull request merged**